### PR TITLE
Bug 1744580: pkg/daemon: create orig file outside of pwd for the file

### PR DESCRIPTION
**Pull request by runcom**
Backing up the original file to its $PWD has a side effect on certain
directories like the Kube static pod dir. If a file is backed up there,
it will be still processed by kube which is completely wrong and
unwanted.
This patch is instead backing up the file inside
/etc/machine-config-daemon/$FPATH avoiding the pitfall above.

Fix https://github.com/openshift/machine-config-operator/issues/1074

Let's debate on the `/etc/machine-config-daemon` location, I don't really like it but I'm accepting other suggestions.

cc @hexfusion 

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

**Comment by cgwalters** [comment]
Oh wow...

/approve

Longer term, I think it'd be a lot better to use ostree for this by default, since that way it's all transactional by default...it would mean we'd be less transactional on "classic RHEL" but hey, everyone running that just accepts that your computer is toast if it crashes in the middle of `yum update` anyways, so...

**Comment by hexfusion** [comment]
LGTM!

/cc @abhinavdahiya @vrutkovs

**Comment by sgreene570** [comment]
@runcom can you fix the typo in your commit message? (Or maybe reword it, the title is kinda confusing).

**Comment by runcom** [comment]
> @runcom can you fix the typo in your commit message? (Or maybe reword it, the title is kinda confusing).

oh yeah, just noticed
